ambry-replication/src/test/java/com.github.ambry.replication/ReplicationTest.java

```diff
@@ -309,6 +321,182 @@ public void replicationPauseTest() throws Exception {
+
+    Transformer transformer = new BlobIdTransformer(storeKeyFactory, storeKeyConverter);
+    ReplicaThread replicaThread =
+        new ReplicaThread("threadtest", replicasToReplicate, new MockFindTokenFactory(), clusterMap,
+            new AtomicInteger(0), localHost.dataNodeId, connectionPool, config, replicationMetrics, null,
+            storeKeyFactory, storeKeyConverter, transformer, clusterMap.getMetricRegistry(), false,
+            localHost.dataNodeId.getDatacenterName(), new ResponseHandler(clusterMap));
+
+
+        /*
+        STORE KEY CONVERTER MAPPING
+        Key     Value
+        B0      B0'
+        B1      B1'
+        B2      null
+
+        BEFORE
+        Local   Remote
+        B0'     B0
+                B1
+                B2
+
+        AFTER
+        Local   Remote
+        B0'     B0
+        B1'     B1
+                B2
+        B0 is B0' for local,
+        B1 is B1' for local,
+        B2 is null for local,
+        so it already has B0/B0'
+        B1 is transferred to B1'
+        and B2 is invalid for L
+        so it does not count as missing
+        Missing Keys: 1
+
+     */
+    Map<PartitionId, List<BlobId>> partitionIdToDeleteBlobId = new HashMap<>();
+    Map<StoreKey, StoreKey> conversionMap = new HashMap<>();
+    Map<PartitionId, BlobId> expectedPartitionIdToDeleteBlobId = new HashMap<>();
+    for (int i = 0; i < partitionIds.size(); i++) {
+      PartitionId partitionId = partitionIds.get(i);
+      List<BlobId> deleteBlobIds = new ArrayList<>();
+      partitionIdToDeleteBlobId.put(partitionId, deleteBlobIds);
+      BlobId b0 = generateRandomBlobId(partitionId);
+      deleteBlobIds.add(b0);
+      BlobId b0p = generateRandomBlobId(partitionId);
+      expectedPartitionIdToDeleteBlobId.put(partitionId, b0p);
+      BlobId b1 = generateRandomBlobId(partitionId);
+      BlobId b1p = generateRandomBlobId(partitionId);
+      BlobId b2 = generateRandomBlobId(partitionId);
+      deleteBlobIds.add(b2);
+      conversionMap.put(b0, b0p);
+      conversionMap.put(b1, b1p);
+      conversionMap.put(b2, null);
+      //Convert current conversion map so that BlobIdTransformer can
+      //create b1p in expectedLocalHost
+      storeKeyConverter.setConversionMap(conversionMap);
+      storeKeyConverter.convert(conversionMap.keySet());
+      addPutMessagesToReplicasOfPartition(Arrays.asList(b0p), Arrays.asList(localHost));
+      addPutMessagesToReplicasOfPartition(Arrays.asList(b0, b1, b2), Arrays.asList(remoteHost));
+      addPutMessagesToReplicasOfPartition(Arrays.asList(b0p, b1), Arrays.asList(null, transformer),
+          Arrays.asList(expectedLocalHost));
+
+      //Check that expected local host contains the correct blob ids
+      Set<BlobId> expectedLocalHostBlobIds = new HashSet<>();
+      expectedLocalHostBlobIds.add(b0p);
+      expectedLocalHostBlobIds.add(b1p);
+      assertEquals("Should be two ids in expectedLocalHostBlobIds", 2, expectedLocalHostBlobIds.size());
```

Comment:
instead I think you want to assert that `expectedLocalHost` has these two blobs? (the size of `expectedLocalHostBlobIds` set is obviously 2).

Reply:
Yeah I can dump that assert.  Yeah it's just ensuring that expectedLocalHost contains what we expect it to contain, no more no less (but without regard to order).